oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexCleaner.java

```diff
@@ -91,7 +91,8 @@ public void run() {
             Set<String> existingIndices = new HashSet<>();
             root.getChildNode(INDEX_DEFINITIONS_NAME).getChildNodeEntries().forEach(childNodeEntry -> {
                 PropertyState typeProperty = childNodeEntry.getNodeState().getProperty(IndexConstants.TYPE_PROPERTY_NAME);
-                if (typeProperty != null && typeProperty.getValue(Type.STRING).equals(ElasticIndexDefinition.TYPE_ELASTICSEARCH)) {
+                String typeValue = typeProperty != null ? typeProperty.getValue(Type.STRING) : "";
+                if (typeValue.equals(ElasticIndexDefinition.TYPE_ELASTICSEARCH) || typeValue.equals("disabled")) {
```

Comment:
Why "typeValue.equals("disabled")"? This included all disabled indexes, including lucene indexes and property indexes... I don't think we should remove those, because they could still be needed. By the way, removing a disabled lucene or property index might take a very long time, and can possibly timeout, because it can contain many descendant nodes.

Reply:
We are not removing any disabled index. In fact this PR is for preventing remote indexes corresponding to locally disabled index from deletion. 
This `if` block is adding `disabled` and `elasticsearch` type indexes which it finds to `existingIndices`. Any existing index is NOT removed from remote elasticsearch server.

In case of disabled lucene and property indexes, the remote index name would be null. Those are not affected by this change.

Reply:
Ah great! Could you add a comment in the code? Otherwise if somebody reads the code in a few years he will not understand...

Reply:
@thomasmueller I have added the comment.